qiskit/algorithms/evolvers/variational/var_qte.py

```diff
@@ -0,0 +1,235 @@
+        )
+
+        linear_solver = VarQTELinearSolver(
+            metric_tensor,
+            evolution_grad,
+            self.lse_solver_callable,
+            self._circuit_sampler,
+            self.imag_part_tol,
+        )
+
+        # Convert the operator that holds the Hamiltonian and ansatz into a NaturalGradient operator
+        ode_function = self.ode_function_factory.build(
+            linear_solver,
+            error_calculator,
+            t_param,
+            init_state_param_dict,
+        )
+
+        ode_solver = VarQTEOdeSolver(
+            init_state_parameters_values, ode_function, self.ode_solver_callable
+        )
+        parameter_values = ode_solver.run(time)
+        param_dict_from_ode = dict(zip(init_state_parameters, parameter_values))
+
+        return initial_state.assign_parameters(param_dict_from_ode)
+
+    def bind_parameters_to_state(
+        self,
+        state: Union[QuantumCircuit, StateFn],
+        param_dict: Dict[Parameter, complex],
+    ) -> None:
+        r"""
+        Bind parameters in a given quantum state to values provided. Uses a ``CircuitSampler`` if
+        available.
+
+        Args:
+            state: Parametrized quantum state to be bound.
+            param_dict: Dictionary which relates parameter values to the parameters in the ansatz.
+        """
+        if self._circuit_sampler:
+            initial_state = self._circuit_sampler.convert(state, param_dict)
+        else:
+            initial_state = state.assign_parameters(param_dict)
+        return initial_state.eval().primitive.data
+
+    @staticmethod
+    def _create_init_state_param_dict(
+        param_value_dict: Dict[Parameter, complex],
+        init_state_parameters: List[Parameter],
+    ) -> Dict[Parameter, complex]:
+        r"""
+        Looks for parameters present in an initial state (an ansatz) in a ``param_value_dict``
+        provided. Based on that, it creates a new dictionary containing only parameters present
+        in an initial state and their respective values. If no value for an initial state parameter
+        is present, it is chosen uniformly at random.
+
+        Args:
+            param_value_dict: Dictionary which relates parameter values to the parameters.
+            init_state_parameters: Parameters present in a quantum state.
+
+        Returns:
+            Dictionary that maps parameters of an initial state to some values.
+        """
+        if param_value_dict is None:
+            init_state_parameter_values = np.random.random(len(init_state_parameters))
+        else:
+            init_state_parameter_values = []
+            for param in init_state_parameters:
+                if param in param_value_dict.keys():
+                    init_state_parameter_values.append(param_value_dict[param])
+                else:
+                    init_state_parameter_values.append(np.random.random(len(init_state_parameters)))
```

Comment:
Should this append a single number or an array? 🤔 Maybe we can also just not support partial parameter dicts, that might be a bit a clearer and simpler behaviour, what do you think?

Reply:
Yeah, I might agree. Partial dicts + partial random inits might be too stealth and lead to errors.